lib/spoom/sorbet/metrics.rb

```diff
@@ -1,90 +1,32 @@
 # typed: strict
 # frozen_string_literal: true
 
+require_relative "sigils"
+
 module Spoom
   module Sorbet
-    class Metrics < T::Struct
+    module Metrics
       extend T::Sig
 
       DEFAULT_PREFIX = "ruby_typer.unknown.."
-      SIGILS = T.let(["ignore", "false", "true", "strict", "strong", "__STDLIB_INTERNAL"], T::Array[String])
-
-      const :raw_metrics, T::Hash[String, T.nilable(Integer)]
 
-      sig { params(path: String, prefix: String).returns(Metrics) }
+      sig { params(path: String, prefix: String).returns(T::Hash[String, Integer]) }
       def self.parse_file(path, prefix = DEFAULT_PREFIX)
         parse_string(File.read(path), prefix)
       end
 
-      sig { params(string: String, prefix: String).returns(Metrics) }
+      sig { params(string: String, prefix: String).returns(T::Hash[String, Integer]) }
       def self.parse_string(string, prefix = DEFAULT_PREFIX)
         parse_hash(JSON.parse(string), prefix)
       end
 
-      sig { params(obj: T::Hash[String, T.untyped], prefix: String).returns(Metrics) }
+      sig { params(obj: T::Hash[String, T.untyped], prefix: String).returns(T::Hash[String, Integer]) }
       def self.parse_hash(obj, prefix = DEFAULT_PREFIX)
-        Metrics.new(
-          raw_metrics: obj["metrics"].each_with_object(Hash.new(0)) do |metric, all|
-            name = metric["name"]
-            name = name.sub(prefix, '')
-            all[name] = metric["value"].to_i
-          end
-        )
-      end
-
-      sig { returns(T::Hash[String, T.nilable(Integer)]) }
-      def files_by_strictness
-        SIGILS.each_with_object({}) do |sigil, map|
-          map[sigil] = raw_metrics["types.input.files.sigil.#{sigil}"]
-        end
-      end
-
-      sig { returns(Integer) }
-      def files_count
-        files_by_strictness.values.compact.sum
-      end
-
-      sig { params(key: String).returns(T.nilable(Integer)) }
-      def [](key)
-        raw_metrics[key]
-      end
-
-      sig { params(out: T.any(IO, StringIO)).void }
-      def show(out = $stdout)
-        files = files_count
-
-        out.puts "Sigils:"
-        out.puts "  files: #{files}"
-        files_by_strictness.each do |sigil, value|
-          next unless value && value > 0
-          out.puts "  #{sigil}: #{value}#{percent(value, files)}"
+        obj["metrics"].each_with_object(Hash.new(0)) do |metric, metrics|
```

Comment:
Basically the only thing remaining here is stripping the keys prefix so it's easier to use in the rest of `spoom`.